{
    "body": "fixes #29226\r\n\r\nThis introduces a new integration test that checks the return code of `spack unit-test` when it is supposed to fail.\r\n\r\nThis is to prevent bugs like the one introduced in #25601 in which CI didn't catch a missing return statement.\r\n\r\nIn retrospective it seems that the shell tests we have right now go through code paths which call `sys.exit` explicitly. This new test instead checks `spack unit-test` which relies on the return code from command invocation in case of errors.",
    "user": "alalazo",
    "url": "https://api.github.com/repos/spack/spack/issues/29244",
    "updated_at": "2022-03-01 06:48:33",
    "created_at": "2022-02-28 11:52:45",
    "closed_at": "2022-02-28 20:55:25",
    "state": "closed",
    "title": "Add a new test to catch exit code failure",
    "number": 29244,
    "milestone": null,
    "labels": [
        "shell-support",
        "maintainers",
        "ci"
    ],
    "id": 1153999572,
    "html_url": "https://github.com/spack/spack/pull/29244",
    "assignees": [],
    "comments": 0
}